testng-core/src/test/kotlin/test/hook/samples/HookSuccessSample.kt

```diff
@@ -0,0 +1,31 @@
+package test.hook.samples
+
+import test.hook.*
+
+import java.util.UUID
+import org.testng.IHookCallBack
+import org.testng.IHookable
+import org.testng.ITestResult
+import org.testng.Reporter
+import org.testng.annotations.DataProvider
+import org.testng.annotations.Test
+import test.TestData
+
+class HookSuccessSample : IHookable {
+
+    override fun run(callBack: IHookCallBack, testResult: ITestResult) {
+        testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true")
+        testResult.setAttribute(HOOK_METHOD_PARAMS_ATTRIBUTE, callBack.parameters)
+        callBack.runTestMethod(testResult)
+    }
+
+    @DataProvider
+    fun dp(): TestData = arrayOf(
+        arrayOf(UUID.randomUUID())
+    )
+
+    @Test(dataProvider = "dp")
```

Comment:
If we are trying a dataprovider test case, the name of the sample must contains dataprovider or provider and have more than 1 data.

Reply:
Wouldn't this rule be relevant to the actual tests rather than samples ? Using a data provider with just 1 set of data is the minimalistic example of a data driven test for which a particular functionality is being vetted out. So can we follow this rule for the actual tests rather than try and apply it for samples ?